modules/apps/frontend-icons/frontend-icons-web/src/main/java/com/liferay/frontend/icons/web/internal/servlet/FrontendIconsServlet.java

```diff
@@ -43,9 +49,10 @@
 	immediate = true,
 	property = {
 		"osgi.http.whiteboard.context.path=/icons",
+		"osgi.http.whiteboard.servlet.name=com.liferay.frontend.icons.web.internal.servlet.FrontendIconsServlet",
 		"osgi.http.whiteboard.servlet.pattern=/icons/*"
 	},
-	service = Servlet.class
+	service = {FrontendIconsServlet.class, Servlet.class}
```

Comment:
Do you really need to export this as `FrontendIconsServlet`? I don't see where it is used...

(Note that I'm reviewing in GitHub, so it is very possible that you are indeed using it, and I have skipped it.)

Reply:
TBH, not sure what is really supposed to go in this field of the OSGI config... I just copied from other examples in DXP

Reply:
I could not found any pattern either. Most of the servlets we have just implemente Servlet.class service. I'd say the other class is needed if service needs to be tracked in a particular way so that one can obtain different instances of the same FrontendIconServlet